### DISCUSSION TITLE
JS 코드 실행과정에서 컴파일과 인터프리트를 활용하는데 어떻게 활용하는 지 설명해주세요.

### DISCUSSION THREAD

**Discussion by wogus216** [diving-javascript/you-dont-know-js-yet#46] [1. 스코프]
JS 코드 실행과정에서 컴파일과 인터프리트를 활용하는데 어떻게 활용하는 지 설명해주세요.

**Comment by lledellebell** [comment]
<details>
<summary>답변</summary>
자바스크립트 엔진은 컴파일과 인터프리트를 혼합하여 사용하여 최적의 성능을 제공합니다. 초기에는 인터프리터를 통해 빠르게 실행을 시작하고, 코드가 자주 실행되면 JIT 컴파일러를 통해 네이티브 코드로 변환하여 성능을 최적화합니다. 
</details>
